business/workloads.go

```diff
@@ -1884,7 +1889,7 @@ func (in *WorkloadService) fetchWorkload(ctx context.Context, criteria WorkloadC
 
 // Get the Waypoint proxy for a workload
```

Comment:
```suggestion
// Get the Waypoint proxies for a workload
func (in *WorkloadService) getWaypointsForWorkload(ctx context.Context, namespace string, workload models.Workload) []models.Workload {
	nsWaypoints, err := in.fetchWorkloads(ctx, namespace, fmt.Sprintf("%s=%s", config.WaypointLabel, config.WaypointLabelValue))
	if err != nil {
		log.Errorf("Error fetching namespace waypoints: %v", err)
		return nil
	}

	var waypoints []models.Workload
	// Get service Account name for each pod from the workload
	for _, nsWaypoint := range nsWaypoints {
		for _, pod := range nsWaypoint.Pods {
			if pod.Labels["istio.io/gateway-name"] == "namespace" {
				waypoints = append(waypoints, *nsWaypoint)
				break
			} else {
				// Get waypoint workloads from a service account
				sa := pod.Annotations["istio.io/for-service-account"]
				for _, pod := range workload.Pods {
					if pod.ServiceAccountName == sa {
						waypoints = append(waypoints, *nsWaypoint)
						break
					}
				}
			}
		}
	}
	return waypoints
}
```

github wouldn't let me do it easily but I am suggesting a replacement for the entire function, because I found it very confusing.   I think this is an example where the comment, function name, and variables can all be improved for readability.  If my suggested  changes are wrong it's probably because I didn't correctly interpret the original function.  But hopefully it's correct, and you find it more clear.